usr/src/cmd/dlmgmtd/dlmgmt_util.c

```diff
@@ -464,7 +464,9 @@ dlmgmt_create_common(const char *name, datalink_class_t class, uint32_t media,
 	linkp->ll_tomb = B_FALSE;
 
 	if (avl_find(&dlmgmt_name_avl, linkp, &name_where) != NULL ||
-	    avl_find(&dlmgmt_id_avl, linkp, &id_where) != NULL) {
+	    avl_find(&dlmgmt_id_avl, linkp, &id_where) != NULL ||
+	    (zoneid == GLOBAL_ZONEID &&
```

Comment:
Should this include a check for `class == DATALINK_CLASS_VNIC`? I don't know how this all hangs together but it concerns me that there is a comment here:
http://src.illumos.org/source/xref/illumos-gate/usr/src/cmd/dlmgmtd/dlmgmt_door.c#44
> * The datalink ID space is global, while each zone has its own datalink name
> * space.  This allows each zone to have complete freedom over the names that
> * they assign to links created within the zone.
> */

We're pretty sure that VNIC names need to be globally unique but I'm not sure about the other classes.
Hopefully someone on #illumos can help/advise.